CMakeLists.txt

```diff
@@ -89,25 +89,18 @@ if(NOT MSVC)
 endif()
 
 if(Boost_FOUND)
-  get_property(SOCI_COMPILE_DEFINITIONS
-    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    PROPERTY COMPILE_DEFINITIONS)
-
-  list(APPEND SOCI_COMPILE_DEFINITIONS "HAVE_BOOST=1")
-  list(APPEND SOCI_COMPILE_DEFINITIONS "BOOST_ALL_NO_LIB")
+  set(SOCI_HAVE_BOOST Y)
```

Comment:
Can we use ON/OFF for Boolean variables as described here https://cmake.org/cmake/help/v3.4/prop_cache/TYPE.html?highlight=bool? I know it is more verbose, but it is also more readable.